rspec-tools/rspec_tools/validation/description.py

```diff
@@ -1,55 +1,160 @@
 from bs4 import BeautifulSoup
 from pathlib import Path
 from typing import Final
+from collections import Counter
 
 from rspec_tools.errors import RuleValidationError
 from rspec_tools.rules import LanguageSpecificRule
 from rspec_tools.utils import LANG_TO_SOURCE
 
 import re
 
+def read_file(path):
+  section_names_path = Path(__file__).parent.parent.parent.parent.joinpath(path)
+  return section_names_path.read_text(encoding='utf-8').split('\n')
+
+def parse_names(path):
+  section_names_path = read_file(path)
+  return [s.replace('* ', '').strip() for s in section_names_path if s.strip()]
+
+HOW_TO_FIX_IT = 'How to fix it'
+HOW_TO_FIX_IT_REGEX = re.compile(HOW_TO_FIX_IT)
+
+def parse_education_section_names(path):
+  education_format_file = read_file(path)
+  sections = {}
+  optional_sections = {}
+  current_map = sections
+  current_section_name = ''
+  for line in education_format_file:
+    if line.startswith('== '):
+      section = line.replace('== ', '').strip()
+      if section.endswith('(optional)'):
+        section = section.replace(' (optional)', '')
+        current_map = optional_sections
+      else:
+        current_map = sections
+      if HOW_TO_FIX_IT_REGEX.match(section):
+        # we store "How to fix it" and "How to fix it in ..." together
+        section = HOW_TO_FIX_IT
+      current_section_name = section
+      if section not in current_map:
+        # If a section appears multiple times in the doc, we don't overwrite its subsections
+        current_map[section] = set()
```

Comment:
```suggestion
        section = HOW_TO_FIX_IT
      else:
        if section in current_map:
          raise RuleValidationError(f'Section {section} is mentioned more than once in the specification file')
      current_section_name = section
      current_map[section] = set()
```
If we don't overwrite its subsections, they are gonna accumulate and be duplicated. 
I think the correct way of handling this is to assert that the section wasn't mentioned before, except if it is a HOW_TO_FIX_IT subsection. See my suggestion there

Reply:
I used sets instead of the initial lists to handle the duplicate subsections.

Reply:
I believe that a check for the tooling itself is a bit overkill, no?